File: 2022/day1/day1.ex
contents = File.read!("input")

group_sums =
  contents
  |> String.trim_trailing("\n")
  |> String.split("\n\n")
  |> Enum.map(&String.split(&1, "\n"))
  |> Enum.map(fn group -> Enum.map(group, &String.to_integer/1) end)
  |> Enum.map(&Enum.sum/1)

max = Enum.max(group_sums)
IO.puts("Max: #{max}")

top3 =
  group_sums
  |> Enum.take(3)
  |> Enum.sum()

IO.puts("Sum of top 3: #{top3}")

File: 2022/day1/day1.rs
use std::fs;
use std::iter::Iterator;

fn main() {
    let contents = fs::read_to_string("input").expect("Input file missing!");
    let groups = split_into_groups(contents);
    let mut sums: Vec<u32> = groups.iter().map(|x| x.iter().sum()).collect::<Vec<u32>>();
    sums.sort_by(|x, y| y.cmp(x));

    println!("Max: {}", sums[0]);
    println!("Sum of top 3: {}", sums.iter().take(3).sum::<u32>())
}

fn split_into_groups(text: String) -> Vec<Vec<u32>> {
    text.split("\n\n")
        .map(|group| {
            group
                .split("\n")
                .map(|int| int.parse::<u32>().unwrap_or(0))
                .collect()
        })
        .collect()
}
